### PR TITLE
Fix max pool forward nhwc

### PR DESCRIPTION
**Pull request by Aidyn-A**
Fixes issue #76432.

Added dilation to loops in CUDA kernel.

cc @ngimel

### CODE REVIEW

**Comment by ngimel** [comment]
Tests please, preferably for a variety of sizes, and for forward and backward preferably. You can add them to test_nn.py

File: test/test_nn.py [hunk 1]
```diff
@@ -16193,6 +16193,53 @@ def helper(n, c, h, w, output_height, output_width, contig):
             helper(4, 8, 9, 14, 5, 8, contig)
             helper(4, 8, 11, 11, 1, 1, contig)
 
+    @dtypes(torch.float, torch.double)
+    def test_pooling_max_nhwc(self, dtype):
```
Comment (ngimel) [inline test/test_nn.py:16197]: tests here automatically get `device` argument, and it's populated with correct device
```suggestion
    def test_pooling_max_nhwc(self, device, dtype):
```

File: test/test_nn.py [hunk 2]
```diff
@@ -16193,6 +16193,53 @@ def helper(n, c, h, w, output_height, output_width, contig):
...
+            self.assertTrue(ref_ind.is_contiguous())
+            self.assertEqual(out, ref_out)
+            self.assertEqual(ind, ref_ind)
+            self.assertEqual(input.grad, ref_input.grad)
+
+        device_list = ['cpu']
+        if TEST_CUDA:
+            device_list.append('cuda')
+
+        for device in device_list:
```
Comment (ngimel) [inline test/test_nn.py:16237]: you won't need this with `device` argument to the test.

**Comment by ngimel** [comment]
Can you also please sign CLA? PR looks good now, let's wait for the tests.

**Comment by Aidyn-A** [comment]
> Can you also please sign CLA? PR looks good now, let's wait for the tests.

I am waiting for corporate CLA to be approved. 
One test is failing. Although, it does not seem to me that it is failing because of changes in this PR

**Review by ngimel — APPROVED** [review]

**Pull request closed without merging**